src/util/funcs.js

```diff
@@ -0,0 +1,8 @@
+export default function formatDuration(courseTime) {
+	const mins = courseTime % 60;
+	const hrs = (courseTime - mins) / 60;
+	if (hrs < 10) {
+		return `0${hrs}:${mins} hours`;
+	}
+	return `${hrs}:${mins} hours`;
+}
```

Comment:
In accordance with the criteria presented in the task, you have a result.
Done:
1. 0.5
2. 0.5
3. 0.5
6.5
7.1.0
8.1.0
9.1.0
10.1.5
12.2
Total: 8.5 out of 14 or 3 out of 5.